src/viking.js

```diff
@@ -1,11 +1,90 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength;
+    }
+    receiveDamage(damage){
+        this.health = this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+    }
+    receiveDamage(damage) {
+        this.health = this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        }
+        return `${this.name} has died in act of combat`;
+    }
+    battleCry() {
+        return 'Odin Owns You All!';
+    }  
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health = this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        }
+        return `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy =[];
+    }
+    
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    }
+
+    vikingAttack() {
+        let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+        let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    
+        let attackedSaxon = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
+    
+        if (this.saxonArmy[randomSaxon].health <= 0) {
+          this.saxonArmy.splice(randomSaxon, 1);
+        }
+        return attackedSaxon;
+      }
+
+    saxonAttack() {
+    let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+
+    let attackedViking = this.vikingArmy[randomViking].receiveDamage(this.saxonArmy[randomSaxon].strength);
+
+    if (this.vikingArmy[randomViking].health <= 0) {
+      this.vikingArmy.splice(randomViking, 1);
+    }
+    return attackedViking;
+  }
+
+    showStatus() {
+        if (this.saxonArmy.length === 0) 
+            return 'Vikings have won the war of the century!';
+        if (this.vikingArmy.length === 0) 
+            return 'Saxons have fought for their lives and survived another day...';
+        return 'Vikings and Saxons are still in the thick of battle.';
+    }
+}
```

Comment:
Nos if que vc colocar o return dessa maneira que fez, vc precisa colocar entre { } assim:
if (this.saxonArmy.length === 0) }
            return 'Vikings have won the war of the century!'
}
e o último return vc pode fazer como fez, mas depois da última chave de um if acima, ele não pode estar solto assim como está
Você pode fazer a mesma lógica que fez, mas com if, else if e o último return como sendo do else (final).
Vê se fica melhor pra vc :)